libs/blocks/text/text.js

```diff
@@ -72,7 +98,7 @@ export default function init(el) {
   });
   rows.forEach((row) => {
     row.classList.add('foreground');
-    decorateBlockText(row, blockTypeSizes[blockType][size]);
+    if (!el.classList.contains('link-farm')) decorateBlockText(row, blockTypeSizes[blockType][size]);
```

Comment:
Creating a const hasLinkFarm and using that whenever required can stop the repetitive queryselect